Import fails for opentelekomcloud_identity_provider_v3

Hi importing opentelekomcloud_identity_provider_v3 does not work

# OTC Provider Version
1.23.9 and 1.23.7

# Terraform Version
0.13.3 and 0.15.1

# Affected Resource(s)
opentelekomcloud_identity_provider_v3

# Steps to Reproduce
terraform import module.otc_keycloak_saml.opentelekomcloud_identity_provider_v3.keycloak playground_keycloak_otc

#Terraform code
module "otc_keycloak_saml" {
  source = "../../../modules/otc_keycloak_saml"
  identity_provider_name = "gotc_keycloak_otc2"
}

resource "opentelekomcloud_identity_provider_v3" "keycloak" {
  name        = var.identity_provider_name
  description = "SSO over Keycloak"
  enabled     = true
}

Creation also does not work

`Plan: 1 to add, 0 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

module.otc_keycloak_saml.opentelekomcloud_identity_provider_v3.keycloak: Creating...
╷
│ Error: Plugin did not respond
│ 
│   with module.otc_keycloak_saml.opentelekomcloud_identity_provider_v3.keycloak,
│   on ../../../modules/otc_keycloak_saml/main.tf line 1, in resource "opentelekomcloud_identity_provider_v3" "keycloak":
│    1: resource "opentelekomcloud_identity_provider_v3" "keycloak" {
│ 
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may contain more details.
╵

Stack trace from the terraform-provider-opentelekomcloud_v1.23.9 plugin:

panic: assignment to entry in nil map`

--- a/opentelekomcloud/services/iam/resource_opentelekomcloud_identity_provider.go
+++ b/opentelekomcloud/services/iam/resource_opentelekomcloud_identity_provider.go
@@ -6,8 +6,9 @@ import (
 	"github.com/hashicorp/go-multierror"
 	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
 	"github.com/opentelekomcloud/gophertelekomcloud"
-	"github.com/opentelekomcloud/gophertelekomcloud/acceptance/clients"
 	"github.com/opentelekomcloud/gophertelekomcloud/openstack/identity/v3/federation/providers"
+
+	"github.com/opentelekomcloud/terraform-provider-opentelekomcloud/opentelekomcloud/common/cfg"
 )
 
 func ResourceIdentityProviderV3() *schema.Resource {
@@ -53,7 +54,8 @@ func ResourceIdentityProviderV3() *schema.Resource {
 }
 
 func resourceIdentityProviderV3Create(d *schema.ResourceData, meta interface{}) error {
-	client, err := clients.NewIdentityV3Client()
+	config := meta.(*cfg.Config)
+	client, err := config.IdentityV3Client(config.GetRegion(d))
 	if err != nil {
 		return fmt.Errorf(clientCreationFail, err)
 	}
@@ -75,7 +77,8 @@ func resourceIdentityProviderV3Create(d *schema.ResourceData, meta interface{}) 
 }
 
 func resourceIdentityProviderV3Read(d *schema.ResourceData, meta interface{}) error {
-	client, err := clients.NewIdentityV3Client()
+	config := meta.(*cfg.Config)
+	client, err := config.IdentityV3Client(config.GetRegion(d))
 	if err != nil {
 		return fmt.Errorf(clientCreationFail, err)
 	}
@@ -105,7 +108,8 @@ func resourceIdentityProviderV3Read(d *schema.ResourceData, meta interface{}) er
 }
 
 func resourceIdentityProviderV3Update(d *schema.ResourceData, meta interface{}) error {
-	client, err := clients.NewIdentityV3Client()
+	config := meta.(*cfg.Config)
+	client, err := config.IdentityV3Client(config.GetRegion(d))
 	if err != nil {
 		return fmt.Errorf(clientCreationFail, err)
 	}
@@ -130,7 +134,8 @@ func resourceIdentityProviderV3Update(d *schema.ResourceData, meta interface{}) 
 }
 
 func resourceIdentityProviderV3Delete(d *schema.ResourceData, meta interface{}) error {
-	client, err := clients.NewIdentityV3Client()
+	config := meta.(*cfg.Config)
+	client, err := config.IdentityV3Client(config.GetRegion(d))
 	if err != nil {
 		return fmt.Errorf(clientCreationFail, err)
 	}

